### PR TITLE
Pass options of entry() and get() to Hyperblobs

### PR DESCRIPTION
**Pull request by LuKks**
Depends on https://github.com/holepunchto/hyperblobs/pull/12 and https://github.com/holepunchto/hypercore/pull/359

### CODE REVIEW

**Comment by gmaclennan** [comment]
This will be very welcome.

The behaviour of `drive.get(path)` though might need some more explanation. Currently the docs say:

> If no blob exists at path, returns null.

However as I understand from testing, there are two other scenarios:

1. If the index (hyperbee) has not downloaded yet, returns `null`
2. If the blobs core (hyperblobs) has not downloaded yet, throw Error 'Block not available'

I'm not sure what the best behaviour is, e.g. when should it throw and when should it return null? Currently throwing because one core has not downloaded, but returning null for the other, does seem a bit confusing.

**Comment by LuKks** [comment]
Tests are not passing as I said on the PR description, it depends on the other one first.

**Comment by LuKks** [comment]
And it also depends on: https://github.com/holepunchto/hypercore/pull/359

**Comment by LuKks** [comment]
@mafintosh Ready!

**Comment by LuKks** [comment]
@gmaclennan Feedback highly appreciated! If you would like to confirm the behaviour about `null` vs `Block not available`, currently it mostly return `null` where it makes sense, let me know!

File: index.js [hunk 1]
```diff
@@ -201,10 +201,15 @@ module.exports = class Hyperdrive extends EventEmitter {
...
-    return typeof name === 'string'
-      ? this.files.get(normalizePath(name))
-      : Promise.resolve(name)
+  async entry (name, opts) {
+    if (typeof name !== 'string') return Promise.resolve(name)
+
+    try {
+      return await this.files.get(normalizePath(name), opts)
+    } catch (err) {
+      if (err.message === 'Block not available locally') return null
```
Comment (mafintosh) [inline index.js:210]: where is this message defined?

Comment (LuKks) [reply index.js:210]: Unrelated, I think we should change this one (removing `0`):
https://github.com/holepunchto/hyperbee/blob/317d565c3431fada1c8e16f71e4465ca4fd92dfd/index.js#L468

Comment (LuKks) [reply index.js:212]: And we should add error codes!

**Comment by LuKks** [comment]
Follow: https://github.com/holepunchto/hyperdrive/pull/348

**Pull request closed without merging**